src/ErrorHandler.php

```diff
@@ -22,7 +22,12 @@ final class ErrorHandler
      *
      * @internal
      */
-    public const DEFAULT_RESERVED_MEMORY_SIZE = 10240;
+    public const DEFAULT_RESERVED_MEMORY_SIZE = 16 * 1024; // 16 KiB
+
+    /**
+     * The regular expression used to match the message of an out of memory error.
+     */
+    private const OOM_MESSAGE_MATCHER = '/^Allowed memory size of (?<memory_limit>\d+) bytes exhausted \(tried to allocate \d+ bytes\)/';
```

Comment:
Are we sure that the message matches on different PHP versions? We should test for that.

Reply:
Well, we are indirectly testing it, because disabling the memory increase (which is the same as a incorrect matcher) fails the added `phpt` test which I think is good enough.

Reply:
It does look like [this string is relatively safe to assume](https://github.com/search?q=repo%3Aphp%2Fphp-src%20%22Allowed%20memory%20size%20of%22&type=code).

[Here is a slightly better regex](https://github.com/php/php-src/blob/524b13460752fba908f88e3c4428b91fa66c083a/Zend/tests/new_oom.phpt#L15) from PHP's source tests.

You may want to use this regex instead.

Reply:
I've updated the regex, only the optional newline was different (`\d+` is functionally identical to `[0-9]+` in this case) and I don't need to match the full string, the first part is enough confirmation it's a OOM error IMHO.